.github/workflows/test-deploy-pnpm.yaml

```diff
@@ -0,0 +1,24 @@
+# Workflow intended to be run only on PRs, that attempts to build locally to catch any errors, using pnpm.
+
+name: Test Deployment - PNPM 
+
+on: 
+  pull_request:
+    # Not specifying a type, lets it run on a PR update 
+    
+jobs:
+  test:
+    runs-on: ubuntu-latest 
+    
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - name: Setup PNPM 
+        run: npm install -g pnpm@v7.0.0
```

Comment:
I really ended up using strict dependencies, just because of issues that can be caused with the latest and greatest in an environment where we want things to work nearly all the time.

I mean unlocked dependencies has already broken the sight once, so I just wanted to avoid it again. As for not using a minor version, there didn't seem to be any features we would benefit that greatly from using this version. And thought that, ideally, the major version would be more stable. But since this is strictly for building and deploying, I'm not that concerned on that aspect. More just ensuring things don't break unexpectedly on us.